### PR TITLE
Chart test

### PR DESCRIPTION
**Pull request by elric1**
From the commit:

Add a chart test framework.

Essentially, the idea is that we have a directory which contains a set of values files: test/charts/*.in.  Corresponding to each chart is the fully expanded chart that they generate.  This is stored in the same directory with the .in replaced with a .expect.  They are both checked into git.

We then provide a script called test-charts which uses "helm template" to expand our charts using each values file and makes sure that it matches the expected chart.  The script will error out if there are any differences.  It will also emit the diffs.

If the diffs are acceptable, just:
```
            :; ./bin/test-charts -r <values_file_name>
```
will put the new expected chart output in place where you can add it to your commit.

We also found a YAML diff program called dyff and add it as an option to helm-compare-versions, we fix a drive-by. And we instrument helm-compare-versions to try to use the test values files if you specify one.

### CODE REVIEW

File: test/charts/mainnet.in [hunk 1]
```diff
@@ -0,0 +1,10 @@
+nodes:
+  country-town:
+    images:
+      octez: tezos/tezos:v10-release
```
Comment (nicolasochem) [inline test/charts/mainnet.in:4]: maybe leave this blank? so we don't have to maintain this file.
Actually, can mainnet.in be empty? Launching the chart with no values at all should work and spin up a mainnet node.

File: test/charts/mainnet.in [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+    runs: [ octez_node, logger, metrics ]
+    instances:
+    - config:
+        shell: {history_mode: rolling}
+  tezos-node: null
+open_acls: true
```
Comment (nicolasochem) [inline test/charts/mainnet.in:10]: this value is outdated.

File: test/charts/private-chain.in [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    quorum_max: 7000
+    quorum_min: 2000
+    seed_nonce_revelation_tip: '125000'
+    time_between_blocks: ['10', '20']
+    tokens_per_roll: '8000000000'
+bootstrap_peers: []
+expected_proof_of_work: 0
+full_snapshot_url: null
+images:
+  tezos: 'tezos/tezos:v10-release'
```
Comment (nicolasochem) [inline test/charts/private-chain.in:48]: should be octez

**Review by nicolasochem — CHANGES_REQUESTED** [review]
This is great, please add a github action to run the test for every PR and push. Thank you.

File: test/charts/private-chain.in (hunk 3 above)
Comment (elric1) [reply test/charts/private-chain.in]: Yes it should! I've fixed it. I decided to have both a mainnet which is empty as you suggested and a mainnet2 which has three nodes in it. I also made the images different at the global and statefulset level in private-chain to make sure that that is being tested.

**Review by nicolasochem — APPROVED** [review]

**Pull request merged**